### ISSUE TITLE
Puzzle Bots (Steam) Black screen after puzzle 2-3

### ISSUE THREAD

**Issue by tag2015** [adventuregamestudio/ags#2147]
**Describe the bug**
After puzzle 2-3 a cutscene should be played, but it never starts. The screen stays black with background music playing.

**AGS Version**
3.6.1.8

**Game**
[Steam link](https://store.steampowered.com/app/80300/Puzzle_Bots/)

**To Reproduce**
1. Load save  [agssave.zip](https://github.com/adventuregamestudio/ags/files/12702891/agssave.zip)
2. Move right

**Desktop (please complete the following information):**
Win10

**Comment by ivan-mogilko** [comment]
Also reproduces with an old version from ~2010 that I had.

**Comment by ivan-mogilko** [comment]
While testing this I found another regression:
When trying to overwrite an existing save, there's a confirmation window, and clicking on "Yes" in that window results in script error:
```
in the active script:
in "GlobalScript.asc", line 2650

Error: Error running function 'btn_over_yes_OnClick':
Error (line 2650): Null pointer referenced
```

This does not happen when running the original exe.

**Comment by ivan-mogilko** [comment]
Technically the issue is that apparently there's a ChangeRoom(12) call from a "room load" event in room 12... So room 12 is being reloaded in an endless loop.

This is a valid behavior, because in AGS same room may be used to make multiple scenes, so it's allowed to reload it again while being inside.

But there might have been some sort of condition to prevent the endless cycle, that is failing. I tried the old AGS where the game was made, and it has same result if I call ChangeRoom(1) in room 1, for instance. There's some factor missing here.

**Comment by ivan-mogilko** [comment]
This is broken by commit a2f9a5ba94377686762b2828df4fccbe661d2b03 .

**Comment by ivan-mogilko** [comment]
Looks like fixed by d7f30763d631c8394c453f82f606eb0fc569e2a1

**Comment by ivan-mogilko** [comment]
I think this should be fixed by 9b1210b5b42503273cf510b5e9b6db34b13f8040

**Comment by tag2015** [comment]
Great job tracking down the problem! I confim that the fix works.
